### ISSUE TITLE
standard deviation of zero during training

### ISSUE THREAD

**Issue by McHaillet** [warpem/miss-alignment#78]
We here just had the problem that MissAlignment crashed because the standard deviation of a subvolume was 0 in the normalization step of the ReconstructionPoolDataset.

**Comment by McHaillet** [comment]
This line: https://github.com/warpem/miss-alignment/blob/9b198dd75379b77c0296c6afbbcad4d3eaa950dc/src/miss_alignment/data/training_dataset.py#L146

Reason is likely that the reconstruction volume dimensions had a lot of empty area in the xy plane. Raw images were roughly 3800x3700, while the provided volume dimensions had 4000x4000 in xy. The error only occured after two succesful macro-iterations with downsampling. It crashed at fine global alignment.

**Comment by McHaillet** [comment]
With too much overhang,  the dimensions should just be limited as a solution. But there could be situations where a strong tilt-axis angle rotation produces empty areas.

We could do:
```python
if std == 0.0:  
    return volume - mean  # this should still work fine
else:
    return (volume - mean) / std
```
Downside is that the model sees flat reconstruction during training, but the confidence head should take care of that.

**Comment by McHaillet** [comment]
Some testers said they worked around it by just having _normalize() return torch.zeros_like(volume) if this condition is hit, and they said the results looked fine. I guess the confidence head should learn to ignore such an example.

**Comment by hamid13r** [comment]
Hi, 

Since we often have various thicknesse, do you think it is a good idea to have conservative z for miss-alignment and go back to a larger value after it is done?

Are you thinking about introducing masks the way isonet and others do at some point?
